tests/references/dune

```diff
@@ -1,3 +1,3 @@
 (library
  (name references)
- (libraries kcas unix))
+ (libraries unix))
```

Comment:
I just decided to drop the two kcas based reference queues as we can use the `Kcas_data.Queue`.